lib/fluent/plugin/filter_process_redfishalert.rb

```diff
@@ -56,13 +59,21 @@ def callRedfishGetAPI(host, resourceURI)
       end
     end
 
-    def getRMCSerialNumber(host)
-      res = callRedfishGetAPI(host, "Chassis/RMC")
+    #we are using nodeID as the unique identifier for dell iDRAC
+    #also, for dell nodeID=SKU=ChassisServiceTag but differs from SN
+    def getMachineIdentifier(host)
+      res = callRedfishGetAPI(host, @hwtDeviceURI[hardware])
+      if @hardware == "Dell_PowerEdge_iDRAC"
+        return res["SKU"]
```

Comment:
You have this hash pattern to get the URL for different hardware types, why not use it for this as well?
something like:
```
return res[@deviceField[hardware]]
```